dashboard/app/models/teacher_score.rb

```diff
@@ -17,4 +17,39 @@
 #
 
 class TeacherScore < ApplicationRecord
+  def self.score_stage_for_section(
+    teacher_id,
+    section_id,
+    stage_id,
+    score
+  )
+    student_ids = Section.find(section_id).students.pluck(:id)
+    level_ids = Stage.find(stage_id).script_levels.map(&:level_id)
+
+    student_ids.each do |student_id|
+      level_ids.each do |level_id|
+        TeacherScore.score_level_for_student(
+          teacher_id,
+          student_id,
+          level_id,
+          score
+        )
+      end
+    end
+  end
+
+  def self.score_level_for_student(
+    teacher_id,
+    student_id,
+    level_id,
+    score
+  )
+    user_level = UserLevel.find_or_create_by(user_id: student_id, level_id: level_id)
```

Comment:
I think we will want to include a script id in this operation, so that we don't get weird behavior when a level in two scripts  has its progress marked in one place but then it gets undone in the other.